apps/ejabberd/src/mod_vcard.erl

```diff
@@ -539,7 +541,7 @@ prepare_vcard_search_params(User, VHost, VCARD) ->
 
     US = {LUser, VHost},
 
-    if
+    case
```

Comment:
Maybe the case could be avoided completely if `b2l` throws a meaningful error when the input is sth else then a binary? Or simply expect badarg when constructing the `#vacard_search` record, this is probably to generic, though.